torch/csrc/jit/codegen/cuda/scheduler/utils.cpp

```diff
@@ -1134,6 +1096,40 @@ IterDomain* projectIdToRoot(
 }
 } // namespace
 
+IterDomain* innerMostRootDim(TensorView* tv) {
+  const auto& root_domain = tv->hasReduction() && tv->hasRFactor()
+      ? tv->getRootDomain()
+      : tv->getMaybeRFactorDomain();
+
+  if (tv->nDims() == 0) {
+    return nullptr;
+  }
+
+  IterDomain* inner_most_id = nullptr;
+
+  for (auto it = root_domain.rbegin(); it != root_domain.rend(); it++) {
+    if ((*it)->isReduction() && tv->isFusionInput()) {
```

Comment:
If we're looking at a reduction domain on an input because of segmentation we don't want to consider those reduction domains as a vectorization opportunity. If we're looking at a reduction reference tensor we want to consider the reduction iteration domains as domains we can vectorize on.

Reply:
Makes sense. Can you please leave it as a comment?